### PR TITLE
Feature/add ability to enable tcpip

### PR DESCRIPTION
**Pull request by dllewellyn**
I couldn't see anything which I was able to use to convert an input stream to a string so I could verify the correct response. Let me know if there is one I can re-use...

### CODE REVIEW

**Comment by vidstige** [comment]
Nice! Great initiative. 👌 Some things:

* You've used the email daniel.llewellyn@discovery.net for committing, but it seems you have not registered this with GitHub. Please use an email you've registred with GitHub for better integration for everyone. You can also register this email if you wish.
* Try to use the _first line_ of the commit message short (~50 chars and max 72). Then two newlines, and then more in-depth information such as links and so on.

* Let's add a unit test for this. You can examine `MockedTestCases.java` first, it should not be too tricky.
* There are no "utility" classes in jadb. The reason is I have worked with _explicitly_ named classes, that helps the reader of the code to understand where to place code. For example the "read all" function provided by the `Utils` class in this PR should be in the `Stream` class, however, such a method already exists. :-)
* The `JadbDevice` class should not use the `inputStream()` method of `Transport`, but instead only use the other functions, in this example  `readResponseTo`. It might be  a bit messy in this case, so feel free to add an `readResponse` method that just returns `String`, e.g. by calling `Stream.readAll(inputStream())`.

Thanks for contributing with this. If you have any questions or need any help of any kind with any part, don't hesitate to contact me. 👍

File: test/se/vidstige/jadb/test/unit/MockedTestCases.java [hunk 1]
```diff
@@ -104,6 +105,14 @@ public void testExecuteShell() throws Exception {
         device.executeShell("ls", "-l");
     }
 
+    @Test
+    public void testExecuteEnableTcpip() throws IOException, JadbException {
+        server.add("serial-123");
+        server.expectTcpip("serial-123", "5555");
+        JadbDevice device = connection.getDevices().get(0);
+        device.enableTcpip();
+    }
```
Comment (vidstige) [inline test/se/vidstige/jadb/test/unit/MockedTestCases.java]: great test 👌

File: test/se/vidstige/jadb/test/fakes/FakeAdbServer.java [hunk 2]
```diff
@@ -173,6 +193,32 @@ public void verifyExpectations() {
...
+
+            @Override
+            public void withContent(byte[] content) {
+
+            }
+
+            @Override
+            public void withContent(String content) {
+            }
+        }
```
Comment (vidstige) [inline test/se/vidstige/jadb/test/fakes/FakeAdbServer.java]: Lots of unused methods here. Lets not inherit from `ExpectationBuilder` just make a brand new class called `TcpIpExpectation` (note that this is an _expectation_, not an _exception_). Then you can remove all the empty functions and just keep the "matches" and the constructor. In fact, at this point, you can just have a list of `int`, each one being a port number where a `enableTcpIp` is expected. 🤷‍♂️ Less code, more win. :-)

File: src/se/vidstige/jadb/JadbDevice.java [hunk 3]
```diff
@@ -139,6 +141,30 @@ private StringBuilder buildCmdLine(String command, String... args) {
...
+     * @param port for the device to bind on
+     *
+     * @return success or failure
+     */
+    public boolean enableTcpip(int port) throws IOException, JadbException {
+        try (Transport transport = getTransport()) {
+            send(transport, String.format("tcpip:%d", port));
+            String expectedResult = String.format("restarting in TCP Mode: %d\n", port);
+            return transport.readString(expectedResult.length()).equals(expectedResult);
+        }
```
Comment (vidstige) [inline src/se/vidstige/jadb/JadbDevice.java]: Hmm, this does not seem right either... 🤔 Do we have to parse the output of this? If the "send". The adb server should respond with a `"OKAY"` any way? E.g. the `send` function will already verify the "OKAY" response. I think we should make this function `void` instead, and just ignore the response string.

File: test/se/vidstige/jadb/test/fakes/FakeAdbServer.java (hunk 2 above)
Comment (dllewellyn) [reply test/se/vidstige/jadb/test/fakes/FakeAdbServer.java]: Great suggestion 👍

**Comment by dllewellyn** [comment]
> Last question: How did you test this?
> 
> * [ ]  Emulator?
> * [ ]  Real device?
> 
> And _how_ did you test it?

No worries - Only on a real device (an emulator doesn't really support adb tcpip) 

Tested with:

(with a device connected over IP) 
```adb shell ipconfig wlan0```

Result:

```
wlan0     Link encap:UNSPEC  
          inet addr:192.168.0.33  Bcast:192.168.0.255  Mask:255.255.255.0 
          inet6 addr: fe80::9e5c:f9ff:fe38:907a/64 Scope: Link
```

Then execute: 

```adb connect 192.168.0.33```

Result:

unable to connect to 192.168.0.33:5555: Connection refused


Then I run:

```
package se.vidstige.jadb;

import java.io.IOException;

public class main {
    public static void main(String[] args) throws IOException, JadbException {
        JadbConnection conn = new JadbConnection();
        for (JadbDevice d : conn.getDevices()) {
            d.enableAdbOverTCP();
        }
    }
}
```

Then try ```adb connect 192.168.0.33``` again (this time it should connect)

**Pull request merged**